c2corg_api/views/user_mailinglists.py

```diff
@@ -0,0 +1,48 @@
+from c2corg_api import DBSession
+from c2corg_api.models.user import User
+from c2corg_api.models.mailinglist import Mailinglist
+from c2corg_api.views import cors_policy, restricted_json_view, \
+    restricted_view
+from c2corg_common.attributes import mailinglists
+from cornice.resource import resource
+from colander import MappingSchema, SchemaNode, String, Boolean, Sequence, \
+    OneOf, required
+from sqlalchemy.orm import joinedload, load_only
+
+
+@resource(path='/users/mailinglists', cors_policy=cors_policy)
+class UserMailinglistsRest(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @restricted_view()
+    def get(self):
+        """Get the mailinglists subscriptions of the authenticated user.
+
+        Request:
+            `GET` `/users/mailinglists`
+
+        Example response:
+
+            {'avalanche': False, 'lawinen': True, 'valanghe': False, ...}
+
+        """
+        user_id = self.request.authenticated_userid
+        listnames = DBSession.query(Mailinglist.listname). \
+            options(joinedload(Mailinglist.user).load_only(User.email)). \
+            filter(User.id == user_id). \
+            all()
+
+        return {ml: ml in mailinglists for ml in listnames}
+
+    @restricted_json_view(schema=MailinglistSchema())
+    def post(self):
+        """Update a mailinglist subscription status.
+
+        Request:
+            `POST` `/users/mailinglists`
+            {'avalanche': False, 'lawinen': True, 'valanghe': False}
```

Comment:
> I plan to pass changed lists into the POST requests with a boolean value to tell if the list must be enabled or disabled for the current user.

What ever works best for you in the UI. You are planning to do a request every time you toggle the subscription for a list? And not send one request containing the status for all lists?

Maybe you don't even need an explicit schema. Simply a validator that checks that the list names are one of `c2corg_common.attributes.mailinglists` could be enough.